### PR TITLE
Display arbitrary alarm times

### LINKED ISSUE #5711: Handle arbitrary alarm durations
As a calendar user, I would like to be able to see my alarms with the correct time, regardless of what duration it is, so I'm not confused by some arbitrary default value.

## AC
- Events that contain alarm triggers that do not correspond to an enum value will fire alarm at the configured time
- In the event editor, it is not possible to set these alarms, but they're displayed and can be cancelled/removed
- The list of alarms on the event is displayed in a human-readable and localized way (5 Stunden instead of 5H)
- only allow single-unit triggers and extend our custom syntax
- incoming triggers with multiple units are converted to the smallest unit they contain (round seconds up to the next full minute)

support weeks, days, hours, minutes.

## Tasks
- [x] Accept arbitrary value when displaying set alarms (probably just the syntax as defined by iCal)
- [x] Format alarm time in a readable and localized way
- [x] Ensure that alarm is still triggered at the right time
   - [x] android
   - [x] ios
   - [x] desktop client
   - [x] web client

## Notes
Example how to convert incoming trigers

-P1DT5M would be 24 * 60 Minutes + 5 Minutes = 1445M

## Test Notes
Test on desktop, web, android, ios:
- [ ] Adding an alarm to a calendar event works (is displayed correctly, bell is on event, is still there when event editor is reopened)
- [ ] Alarm triggers at the correct time

Does not need to be tested on all platforms:
- [ ] Importing an event with an alarm works as expected (alarm is on the event, triggers at the correct time)
- [ ] Importing an event with multiple time units on the trigger resolves to the smallest unit given (1DT30M = 1470 minutes, 2W1D = 15 days, etc...) and that value is displayed in the reminder time
- [ ] The minutes/days/weeks label matches the user's chosen language (Minuten/Tage/Wochen in German, for example)

### PR DESCRIPTION
**Pull request by wrdhub**
Handle arbitrary alarm times for web/desktop, including native changes on ios and android

close #5711

### CODE REVIEW

File: src/calendar/export/CalendarParser.ts [hunk 1]
```diff
@@ -266,72 +267,59 @@ function parseAlarm(alarmObject: ICalObject, event: CalendarEvent): AlarmInfo |
 	}
 	const triggerValue = triggerProp.value
 	if (typeof triggerValue !== "string") throw new ParserError("expected TRIGGER property to be a string: " + JSON.stringify(triggerProp))
-	let trigger: AlarmInterval
+	let alarmInterval: AlarmInterval
 
 	// Absolute time
 	if (triggerValue.endsWith("Z")) {
+		// For absolute time we just convert the trigger to minutes. There might be a bigger unit that can express it but we don't have to take care about time
```
Comment (ganthern) [inline src/calendar/export/CalendarParser.ts:274]: can we normalize zero-length intervals to always use the same unit?

**Review by ganthern — COMMENTED** [review]
* we lost the deduplication of alarm intervals even though nothing about creating them had to change. would be nice to have an explanation at least.
* there's no tests for the parsing of multi-unit ical triggers as far as I can see?

File: src/calendar/export/CalendarParser.ts (hunk 1 above)
Comment (wrdhub) [reply src/calendar/export/CalendarParser.ts:274]: I guess we could. Is that something we want to do?

File: src/calendar/export/CalendarImporter.ts [hunk 2]
```diff
@@ -222,35 +222,18 @@ function makeInvitationCalendar(versionNumber: string, event: CalendarEvent, met
...
-		case AlarmInterval.THREE_DAYS:
-			return "-P3D"
-
-		case AlarmInterval.ONE_WEEK:
-			return "-P1W"
-
-		default:
-			throw new Error("unknown alarm interval: " + alarmInterval)
+	if (alarmInterval.unit === AlarmIntervalUnit.MINUTE || alarmInterval.unit === AlarmIntervalUnit.HOUR) {
+		serialized += "T" + pad2(alarmInterval.value) + alarmInterval.unit
```
[the remark this answers is not in the document]
Comment (wrdhub) [reply src/calendar/export/CalendarImporter.ts:231]: You're right, for some reason we thought it was.

File: src/calendar/export/CalendarParser.ts (hunk 1 above)
Comment (ganthern) [reply src/calendar/export/CalendarParser.ts:274]: It's probably not important since we don't deduplicate them anymore anyway

**Pull request merged**